spec/migrations/20190118204247_move_miq_set_memberships_spec.rb

```diff
@@ -0,0 +1,36 @@
+require_migration
+
+describe MoveMiqSetMemberships do
+  let(:rel_stub) { migration_stub(:Relationship) }
+  let(:membership_stub) { migration_stub(:MiqSetMembership) }
+  let(:ns_stub) { Class.new(ActiveRecord::Base) { self.table_name = "miq_ae_namespaces" } }
+  let(:all_set_memberships) { membership_stub.all }
+
+  migration_context :up do
+    context "parent/child/grandchild rel" do
+      it 'updates ancestry' do
+        ns1 = ns_stub.create!
+        ns1_child = ns_stub.create!
+        ns1_grandchild = ns_stub.create!
+        ns1_rel = rel_stub.create!(:relationship => "membership", :ancestry => nil, :resource_id => ns1.id, :resource_type => "MiqAeNamespace")
+        ns1_child_rel = rel_stub.create!(:relationship => "membership", :ancestry => ancestry_for(ns1_rel), :resource_id => ns1_child.id, :resource_type => "MiqAeNamespace")
+        rel_stub.create!(:relationship => "membership", :ancestry => ancestry_for(ns1_rel, ns1_child_rel), :resource_id => ns1_grandchild.id, :resource_type => "MiqAeNamespace")
```

Comment:
this is pretty similar to the code in a couple of PRs.
and this is a simple linear 3 level relationship.

going line by line it looks good.
but I'm having trouble parsing it.

Also, on the note of contract tests, I see how you are creating these, but I'm not sure that this matches the way that it is coded elsewhere.
I do seem to remember a linear only 1 parent, 1 child model.

this seems to be different from the way it was stated above (`we only create these for the first level`).
sorry, being vague here - just a little confused and the create rel create ns isn't making it any easier for my brain.

Reply:
it was intending as a first pass test to show that that my statement about how `we only create these for the first level` was in fact veridical.